sentry_streams/sentry_streams/pipeline/message.py

```diff
@@ -0,0 +1,27 @@
+from __future__ import annotations
+
+from typing import (
+    Any,
+    Generic,
+    MutableMapping,
+    Optional,
+    TypeVar,
+)
+
+from sentry_kafka_schemas.codecs import Codec
+
+TIn = TypeVar("TIn")
+
+
+# a message with a generic payload
+class Message(Generic[TIn]):
```

Comment:
I am trying to intentionally keep Message as bare-bones as possible. For example, I don't think the user should have to think about:

headers after the source step
partition key
offset management
However, they can provide a schema for messages and any additional info. e.g. for the generic metrics pipeline, maybe they annotate each message with the metric type.

Do let me know if something important is missing though.

Reply:
> Expose headers and separate them between internal ones managed by platform, mutable by the platform and the application ones: readonly and have to be present at the source. We will figure out later how to provide a more flexible set of application headers

With tasks, we have headers as as simple mutable map. Separating the headers out seems like additional complexity we could avoid. Our SDKs use message headers to pass trace_id and trace baggage which I assume you'll want for streaming as well.

Reply:
for the time being: add headers as a map, propagate in all steps, leave empty in reduce.
Add timestamp.